deep_translator/base.py

```diff
@@ -128,6 +132,23 @@ def translate(self, text: str, **kwargs) -> str:
         """
         return NotImplemented("You need to implement the translate method!")
 
+    @abstractmethod
+    @lru_cache(maxsize=128)
```

Comment:
Would be better if you make the cache_maxsize value in a separate global config.py file